tests/unit/core/test_utils.js

```diff
@@ -495,23 +526,72 @@ describe('isCompatibleWithUserAgent', () => {
       browser: { name: 'Firefox', version: '54.0' },
       os: { name: 'Windows' },
     };
-    expect(isCompatibleWithUserAgent({ _log: fakeLog, addon: fakeAddon, minVersion: '*', userAgentInfo }))
+    expect(isCompatibleWithUserAgent({
+      _log: fakeLog,
+      addon: createInternalAddon(fakeAddon),
+      minVersion: '*',
+      userAgentInfo,
+    }))
       .toEqual({ compatible: false, reason: INCOMPATIBLE_UNDER_MIN_VERSION });
     expect(fakeLog.error.firstCall.args[0])
       .toContain('minVersion of "*" was passed to isCompatibleWithUserAgent()');
   });
 
   it('is incompatible with empty user agent values', () => {
     const userAgentInfo = { browser: { name: '' } };
-    expect(isCompatibleWithUserAgent({ addon: fakeAddon, userAgentInfo }))
+    expect(isCompatibleWithUserAgent({
+      addon: createInternalAddon(fakeAddon), userAgentInfo,
+    }))
       .toEqual({ compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX });
   });
 
   it('is incompatible with non-string user agent values', () => {
     const userAgentInfo = { browser: { name: null }, os: { name: null } };
-    expect(isCompatibleWithUserAgent({ addon: fakeAddon, userAgentInfo }))
+    expect(isCompatibleWithUserAgent({
+      addon: createInternalAddon(fakeAddon), userAgentInfo,
+    }))
       .toEqual({ compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX });
   });
+
+  it('is incompatible if no matching platform file exists', () => {
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      current_version: {
+        ...fakeAddon.current_version,
+        files: [{
+          ...fakeAddon.current_version.files[0],
+          platform: OS_MAC,
+        }],
+      },
+    });
+    const userAgentInfo =
+      UAParser(userAgentsByPlatform.windows.firefox40);
+
+    expect(isCompatibleWithUserAgent({ addon, userAgentInfo }))
+      .toEqual({
+        compatible: false,
+        reason: INCOMPATIBLE_UNSUPPORTED_PLATFORM,
+      });
+  });
+
+  it('allows non-extensions to have mismatching platform files', () => {
```

Comment:
Does this ever happen? I didn't think themes could/did set platform.

Reply:
I'm not sure but since this code is only working around a bug in extensions I wanted to limit it there until we know we need it elsewhere.